File: KalmanFilter/KalmanTestMain.m
%% Input data here
close all

data = csvread('/Users/celialewis/Dropbox/SeniorDesignData/raw_player-April-22-2018-1301.csv');

avgx = mean(data(:,1));
variancex = var(data(:,1));
stand_devx = sqrt(variancex);
 
avgy = mean(data(:,2));
variancey = var(data(:,2));
stand_devy = sqrt(variancey);
 
for i = 11:length(data(:,1))
    if data(i,1) >= avgx+1*stand_devx || data(i,1) <= avgx-1*stand_devx
        data(i,1) = data((i-1),1);
    end
    if data(i,2) >= avgy+1*stand_devy || data(i,2) <= avgy-1*stand_devy
        data(i,2) = data((i-1),2);
    end
end

f = figure; 

GenerateKalmanFilt(data(:,1), avgx, variancex, f, 1);
title('X Position Kalman Output')
GenerateKalmanFilt(data(:,2), avgy, variancey, f, 2);
title('Y Position Kalman Output')

File: KalmanFilter/KalmanTestMain.py
import csv
import numpy as np
from GenerateKalmanFilt import *

data = []

path = '/Users/BenFrancis/Dropbox/SeniorDesignData/raw_player-April-22-2018-1301.csv'

with open(path, 'rt') as csvfile:
    reader = csv.reader(csvfile, delimiter=',')
    for row in reader:
        data.append(row)

xData = list(map(lambda x: float(x[0]), data))
yData = list(map(lambda x: float(x[1]), data))


xAvg = np.mean(xData)
xVar = np.var(xData)
xStd = np.std(xData)

yAvg = np.mean(yData)
yVar = np.var(yData)
yStd = np.std(yData)


for i in range(11, len(xData)):
    if xData[i] >= xAvg + xStd or xData[i] <= xAvg - xStd:
        xData[i] = xData[i-1]
    if yData[i] >= yAvg + yStd or yData[i] <= yAvg - yStd:
        yData[i] = yData[i-1]


print(xData)
GenerateKalmanFilt(xData, xAvg, xVar)
